diff --git a/CMakeLists.txt b/CMakeLists.txt
index c06124c93..4630da312 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -288,10 +288,10 @@ IF (WIN32)
   SET_PROPERTY(GLOBAL PROPERTY USE_FOLDERS ON)
 
   IF ((${CMAKE_CXX_COMPILER_ID} MATCHES "IntelLLVM") OR
-      (${CMAKE_CXX_COMPILER_ID} MATCHES "Clang" AND COMPILER_HAS_SYCL_SUPPORT))
+      (${CMAKE_CXX_COMPILER_ID} MATCHES "Clang" AND COMPILER_HAS_SYCL_SUPPORT AND NOT EMBREE_ARM))
     MESSAGE("-- DPCPP compiler detected")
     INCLUDE(dpcpp)
-  ELSEIF(${CMAKE_GENERATOR_TOOLSET} MATCHES "^LLVM")
+  ELSEIF(${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
     set(EMBREE_SYCL_SUPPORT OFF)
     MESSAGE("-- CLANG compiler detected")
     INCLUDE(clang)
diff --git a/common/cmake/FindTBB.cmake b/common/cmake/FindTBB.cmake
index ed7db34db..b54f0bd70 100644
--- a/common/cmake/FindTBB.cmake
+++ b/common/cmake/FindTBB.cmake
@@ -292,7 +292,9 @@ function(rk_tbb_find_library COMPONENT_NAME BUILD_CONFIG)
   unset(LIB_PATHS)
 
   if (WIN32)
-    if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+    if(CMAKE_SYSTEM_PROCESSOR MATCHES "ARM64|aarch64|AARCH64")
+      set(TBB_ARCH arm64)
+    elseif(CMAKE_SIZEOF_VOID_P EQUAL 8)
       set(TBB_ARCH intel64)
     else()
       set(TBB_ARCH ia32)
@@ -358,7 +360,7 @@ function(rk_tbb_find_library COMPONENT_NAME BUILD_CONFIG)
   endif()
 
   find_library(${LIB_VAR}
-    NAMES ${LIB_NAME}
+    NAMES ${LIB_NAME} ${LIB_NAME_VERSION}
     PATHS ${LIB_PATHS}
     NO_DEFAULT_PATH)